src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -404,6 +404,16 @@ void FlightPathMovementGenerator::DoFinalize(Player* player)
     }
 
     player->RemoveFlag(PLAYER_FLAGS, PLAYER_FLAGS_TAXI_BENCHMARK);
+
+    if (ShapeshiftForm form = player->GetShapeshiftForm())
+    {
+        SpellShapeshiftFormEntry const* shapeshift = sSpellShapeshiftFormStore.LookupEntry(form);
+        if (!shapeshift)
```

Comment:
Why is this check needed?

Reply:
To check if the player is shapeshifted to return his displayid

Reply:
But player->GetShapeshiftForm() already gives you all thats needed

Reply:
Fixed